utils/src/main/kotlin/Utils.kt

```diff
@@ -143,20 +143,25 @@ fun filterVersionNames(version: String, names: List<String>, project: String? =
 
 /**
  * Check if the "user.home" property is set to a sane value and otherwise set it to the value of an (OS-specific)
- * environment variable for the user home directory. This works around the issue that esp. in certain Docker scenarios
- * "user.home" is set to "?", see https://bugs.openjdk.java.net/browse/JDK-8193433 for some background information.
+ * environment variable for the user home directory, and return that value. This works around the issue that esp. in
+ * certain Docker scenarios "user.home" is set to "?", see https://bugs.openjdk.java.net/browse/JDK-8193433 for some
+ * background information.
  */
-inline fun <reified T : Any> T.fixupUserHomeProperty() {
+inline fun <reified T : Any> T.fixupUserHomeProperty(): String {
     val userHome = System.getProperty("user.home")
-    if (userHome.isNullOrBlank() || userHome == "?") {
-        listOfNotNull(
-            Os.env["HOME"],
-            Os.env["USERPROFILE"]
-        ).firstOrNull { it.isNotBlank() }?.let {
-            log.info { "Fixing up the user home directory from '$userHome' to '$it'." }
-            System.setProperty("user.home", it)
-        }
+    if (!userHome.isNullOrBlank() && userHome != "?") return userHome
+
+    val fallbackUserHome = listOfNotNull(
+        Os.env["HOME"],
+        Os.env["USERPROFILE"]
+    ).first {
+        it.isNotBlank()
```

Comment:
I think we should rather throw a custom exception here explaining the problem instead of the generic `NoSuchElementException`. I'm afraid users might otherwise have a hard time understanding the problem.

Reply:
Done.